starknet_devnet/dump.py

```diff
@@ -0,0 +1,43 @@
+"""Dumping utilities."""
+
+import multiprocessing
+
+import dill as pickle
+
+from .util import DumpOn
+
+def perform_dump(dumpable, path):
+    """Performs the very act of dumping."""
+    with open(path, "wb") as file:
+        pickle.dump(dumpable, file)
+
+class Dumper:
+    """Class for dumping objects."""
+
+    def __init__(self, dumpable):
+        """Specify the `dumpable` object to be dumped."""
+
+        self.dumpable = dumpable
+
+        self.dump_path: str = None
+        """Where to dump."""
+
+        self.dump_on: DumpOn = None
+        """When to dump."""
+
+    def dump(self, path: str):
+        """Dump to `path`."""
+        path = path or self.dump_path
+        assert path, "No dump_path defined"
+        print("Dumping Devnet to:", path)
+
+        multiprocessing.Process(
+            target=perform_dump,
+            args=[self.dumpable, path]
+        ).start()
+        # don't .join(), let it run in background
+
+    def dump_if_required(self):
```

Comment:
I'm not sure about this method. What if new `dump_on` trigger are added?

Reply:
I was also very unsure, but what other options are there? The logic could be inside the add_transaction endpoint.

Reply:
Perhaps to have `dump_on` as an argument?

Reply:
Addressed it here: https://github.com/Shard-Labs/starknet-devnet/pull/40/commits/67fe6aa3efbd7011ce9905ea299baed0b0a4c576